kernel/src/table_properties/deserialize.rs

```diff
@@ -0,0 +1,284 @@
+    }
+}
+
+/// Deserialize a comma-separated list of column names into an `Option<Vec<ColumnName>>`.
+pub(crate) fn deserialize_column_names<'de, D>(
+    deserializer: D,
+) -> Result<Option<Vec<ColumnName>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    Ok(Some(
+        ColumnName::parse_column_name_list(&s).map_err(de::Error::custom)?,
+    ))
+}
+
+/// Deserialize an interval string of the form "interval 5 days" into an `Option<Duration>`.
+pub(crate) fn deserialize_interval<'de, D>(deserializer: D) -> Result<Option<Duration>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    parse_interval(&s).map(Some).map_err(de::Error::custom)
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum ParseIntervalError {
+    /// The input string is not a valid interval
+    #[error("'{0}' is not an interval")]
+    NotAnInterval(String),
+    /// Couldn't parse the input string as an integer
+    #[error("Unable to parse '{0}' as an integer")]
+    ParseIntError(String),
+    /// Negative intervals aren't supported
+    #[error("Interval '{0}' cannot be negative")]
+    NegativeInterval(String),
+    /// Unsupported interval
+    #[error("Unsupported interval '{0}'")]
+    UnsupportedInterval(String),
+    /// Unknown unit
+    #[error("Unknown interval unit '{0}'")]
+    UnknownUnit(String),
+}
+
+/// This is effectively a simpler version of spark's `CalendarInterval` parser. See spark's
+/// `stringToInterval`:
+/// https://github.com/apache/spark/blob/5a57efdcee9e6569d8de4272bda258788cf349e3/sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/SparkIntervalUtils.scala#L134
+///
+/// Notably we don't support months nor years, nor do we support fractional values, and negative
+/// intervals aren't supported.
+///
+/// For now this is adapted from delta-rs' `parse_interval` function:
+/// https://github.com/delta-io/delta-rs/blob/d4f18b3ae9d616e771b5d0e0fa498d0086fd91eb/crates/core/src/table/config.rs#L474
+///
+/// See issue delta-kernel-rs/#507 for details: https://github.com/delta-io/delta-kernel-rs/issues/507
+fn parse_interval(value: &str) -> Result<Duration, ParseIntervalError> {
+    let mut it = value.split_whitespace();
+    if it.next() != Some("interval") {
+        return Err(ParseIntervalError::NotAnInterval(value.to_string()));
+    }
+    let number = it
+        .next()
+        .ok_or_else(|| ParseIntervalError::NotAnInterval(value.to_string()))?;
+    let number: i64 = number
+        .parse()
+        .map_err(|_| ParseIntervalError::ParseIntError(number.into()))?;
+
+    // TODO(zach): spark allows negative intervals, but we don't
+    if number < 0 {
+        return Err(ParseIntervalError::NegativeInterval(value.to_string()));
+    }
+    let number: u64 = number
+        .try_into()
+        .expect("interval is non-negative and was i64, so it fits in u64");
```

Comment:
Isn't this a panic? 
Since we already proved it's safe, can we not just cast?
```suggestion
    let number = number as u64; // non-negative i64 => always safe
```
(but do we actually _want_ u64 given that (a) java doesn't support it; and (b) we may need to support negative intervals some day?)

Reply:
well we need u64 since duration API expects that. i'll leave the cast here (updated with yours) with a note